src/crystal/system/win32/delay_load.cr

```diff
@@ -15,6 +8,21 @@ private macro p_from_rva(rva)
   pointerof(LibC.image_base).as(UInt8*) + {{ rva }}
 end
 
+private macro print_error(format, *args)
+  {% if args.empty? %}
+    %str = {{ format }}
+    LibC.WriteFile(LibC.GetStdHandle(LibC::STD_ERROR_HANDLE), %str, %str.bytesize, out _, nil)
+  {% else %}
+    %buf = uninitialized LibC::CHAR[1024]
+    %args = uninitialized UInt64[{{ args.size }}]
+    {% for arg, i in args %}
+      %args[{{ i }}] = {{ arg }}
+    {% end %}
+    %len = LibC.FormatMessageA(LibC::FORMAT_MESSAGE_FROM_STRING | LibC::FORMAT_MESSAGE_ARGUMENT_ARRAY, {{ format }}, 0, 0, %buf, %buf.size, %args)
```

Comment:
I think it's really confusing that args must each be an integer already at call site. Why not move the `.address` call into the macro?

Reply:
I'm not sure that would be less confusing. It's about printing the pointer's address, not its value. So converting to the address at call site seems reasonable.

Reply:
You either do this or create a pointer from the ordinal at the call site. The latter is probably better (`DWORD_PTR` and `ULONG_PTR` are really pointer-sized)